### ISSUE TITLE
Msys not handling correclty Windows users

### ISSUE THREAD

**Issue by JDarthenay** [Alexpux/MSYS2-pacman#32]
I am not sure I am opening this issue in the right repository.

I am opening this issue because it seems to me the installer is setup everything as if Windows is a single user system. Well, maybe a lot of people are doing that, but this is a wrong way to do things and no software should encourage this (dumb) behavior.

It seems the installer does a single user installation because:
- shortcuts are added in installing user menu rather than "all users" menu.
- when running "mingw64_shell.bat" with non administrator user, I keep getting chmod errors because it cannot change permissions of '/dev/shm' and '/dev/mqueue'. When I look authorizations in the matching Windows folders it seem to me normal I cannot do that with my current Windows user.

So, I propose to fix things by implementing the following enhancements:
- do not default to single user installation. Either give the choice between single user installation and all users installation, or provide only all users installation.
- if single user installation is possible, it should not require admin privileges. Elevation prompt should occur only once you select all users installation. If elevation prompt cannot be shown after start, so the choice between single user and all users installation should be done by selecting another installer, and the single user installer should not show elevation prompt.
- after a all user installation, running MSYS2 prompts should not requires write access to common files or folders. Each user should write only his own files.
- after a all user installation, only a user with admin privileges should be able to run pacman.

Fixing this issue seems mandatory to me. So far I used MSYS2 only to download packages, and I won't do more if this issue is not solved.

If you lack time to solve it, I may try to see if I can participate to MSYS2 development and fix it.

**Comment by mingwandroid** [comment]
https://github.com/Alexpux/MSYS2-packages is the correct place for this issue since it'll come down to changing https://github.com/Alexpux/MSYS2-packages/tree/master/msys2-installer. Your contributions would be welcomed but it needs detailed discussion. Please open the issue on that project so that subscribers to it get notifications to the discussion. There probably aren't many subscribers to this project.

**Comment by JDarthenay** [comment]
https://github.com/Alexpux/MSYS2-packages/issues/557
I guess you can close this one.
